CMakeLists.txt

```diff
@@ -64,6 +64,10 @@ option(PRECICE_ALWAYS_VALIDATE_LIBS "Validate libraries even after the validatat
 option(PRECICE_ENABLE_C "Enable the native C bindings" ON)
 option(PRECICE_ENABLE_FORTRAN "Enable the native Fortran bindings" ON)
 
+option(PRECICE_RELEASE_WITH_DEBUG_LOG "Enable debug logging in release builds" OFF)
```

Comment:
This potentially slows down simulations significantly. Normally, users would expect to get a production version from installing a package, no? I get your point, but I am not sure this is a good solution. How would we then tell users: "actually you should not run this version, but the fast one".
I think we just have to be a bit more relaxed if users come saying "my simulation crashes". This can have many reasons and only very few of them are actual bugs in preCICE, where debug output would help.

Reply:
Everything that is not a bug, but "a feature", one should be able to analyze with INFO output (and convergence files etc).

Reply:
I see your point (assuming that this indeed significantly slows down, I assumed that the different levels also help this situation to some extent) and I agree that we should not change it in that case.

I guess it is not as common anymore that we find bugs in preCICE itself, but then I wonder a bit how to "sell" this feature. For the VM it would indeed not be a problem to have slower simulations and maybe we could ask users to reproduce their problems in the VM (or in a debug build on their system).

Reply:
> I guess the feature is mainly for us and the VM.

This is also useful if an adapter developer needs to debug a case with uses connectivity.
Of course this includes ASTE. During the tests of the performance paper, we ran into issues where the Release build would run into issues and the Debug build would timeout while setting the connectivity.